crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -56,6 +59,21 @@ impl<'w, E, B: Bundle> Trigger<'w, E, B> {
     pub fn entity(&self) -> Entity {
         self.trigger.entity
     }
+
+    /// Enables event bubbling.
+    pub fn propagate(&mut self, should_bubble: bool) {
+        *self.bubble = should_bubble;
+    }
```

Comment:
I could be wrong but I think "bubbling" is a subset of "propagation" (specifically one that goes _up_ a hierarchy). It might be better to keep these two terms separate from one another, and have the parameter be `should_propagate`.

Or we could also consider moving away from the web-based terminology altogether and call `bubbles`/`should_bubble` something like `traverses`/`should_traverse`?

Reply:
Good point. I'm good with whatever people like best. The terminology here is ported pretty directly from `bevy_eventlistener` which mixes the two.

Reply:
I prefer propagation for the generic term, and bubbling for the "walk up the hierarchy" behavior specifically.

Reply:
> I prefer propagation for the generic term, and bubbling for the "walk up the hierarchy" behavior specifically.

One potential issue is that we don't control which way we're walking. If a library implements `Event` where `Traverse` is `MainChild` or something, then we're technically walking down the tree.

This might get even more confusing if we introduce a second phase to the propagation (like a "capture" phase). It might happen that the way a particular event is set up that things are seemingly swapped to the consumer of that event.

Not sure how likely this all is but worth thinking about imo.

Maybe we remove the term "bubble" entirely and prefer "traverse" in its place? Or maybe we rename the assoc type to `Bubble` to indicate the intended direction (I guess that's assuming we ever plan on adding a second phase). Or maybe it's not that big a deal and we just leave it as is haha

Reply:
> In any case, my point is don't spend any extra effort enabling a use case which might never happen.

I agree with this. I'd be ok actually going the other way and making the `Traversal` trait more specific to upward propagation/bubbling. The relations working group have told me there's little chance this can be forward compatible with relations, so there's no point trying to broaden the scope too much.

Reply:
> > In any case, my point is don't spend any extra effort enabling a use case which might never happen.
> 
> 
> 
> I agree with this. I'd be ok actually going the other way and making the `Traversal` trait more specific to upward propagation/bubbling.

I could be completely misinterpreting this but I feel like naming it something more specific to the directionality is more for the purpose of enabling future behavior in alternative directions. If we want to avoid the issue altogether then I feel like a general `traverse`/`propagate` term might be clearer in that case rather than doubling on the bubbling (sorry lol)

Reply:
I am fine with any of it but I really would like to get this cleaned up for approval as soon as possible. @alice-i-cecile how do we feel about keeping the trait as "traversal" and changing the observers/events part to always use "bubbling" over "propagation"?

Reply:
Sounds fine 👍🏽

Reply:
Ended up going with Propagation everywhere, since `Bubbling` is more of a `bevy_hierarchy` concept.